torch/_decomp/decompositions.py

```diff
@@ -847,6 +847,70 @@ def native_group_norm(
     return (out, mean, rstd)
 
 
+@register_decomposition(aten.native_group_norm_backward)
+def native_group_norm_backward(
+    grad_output: Tensor,
+    input: Tensor,
+    mean: Tensor,
+    rstd: Tensor,
+    gamma: Optional[Tensor],
+    N: int,
+    C: int,
+    HxW: int,
+    group: int,
+    output_mask: List[bool],
+) -> Tuple[Optional[Tensor], Optional[Tensor], Optional[Tensor]]:
+    assert grad_output.shape == input.shape
+    assert input.numel() == N * C * HxW
+    assert mean.shape == (N, group)
+    assert rstd.shape == (N, group)
+    assert gamma is None or gamma.numel() == C
+
+    # Compute Internal gradients
+    acc_dtype = utils.get_computation_dtype(input.dtype)
+    ds = torch.mul(grad_output, input.to(dtype=acc_dtype)).view(N, C, HxW).sum(dim=[2])
+    db = grad_output.view(N, C, HxW).sum(dim=[2], dtype=acc_dtype)
+
+    d_input: Optional[Tensor] = None
+    d_gamma: Optional[Tensor] = None
+    d_bias: Optional[Tensor] = None
+    cpg = C // group
+    if output_mask[0]:
+        s = 1.0 / (HxW * cpg)
+        if gamma is not None:
+            ds_val = torch.mul(ds, gamma.reshape(1, C)).reshape(N, group, cpg).sum(2)
```

Comment:
`gamma.unsqueeze(0)`, `unsqueeze` is typically easier for backends to reason about